generation/localizedstringkit/detection.py

```diff
@@ -197,32 +197,151 @@ def _detect_strings(
+        r'LocalizedWithKeyExtensionAndBundle\(\s*(?:#*"(.*?)"#*|"(.*?)")\s*,\s*(?:#*"(.*?)"#*|"(.*?)")\s*,\s*(?:#*"(.*?)"#*|"(.*?)")\s*,\s*(?:#*"(.*?)"#*|"(.*?)")\s*\)', re.DOTALL
     )
 
+    def confirm_string_args_only(self) -> None:
+        """Confirm that there are only string arguments passed in for Swift files.
+
+        Overrides the base class method to allow raw strings (#"..."#) in addition to regular strings.
+
+        :raises InvalidLocalizedCallException: If there are invalid calls
+        """
+        matches = SwiftDetector._SWIFT_INVALID_CALL_PATTERN.findall(self.sanitized_contents)
+
+        # Strip spaces
+        matches = [match.strip() for match in matches]
+
+        # Filter out redefinitions
+        matches = [match for match in matches if not match.startswith("func ")]
+
+        if len(matches) > 0:
+            raise InvalidLocalizedCallException(
+                f"Found invalid calls to Localized in file: {self.file_path}, {matches}"
+            )
+
+    def _extract_string_content(self, raw_capture: str, regular_capture: str) -> str:
+        """Extract the actual string content from raw or regular string captures.
+
+        :param raw_capture: The capture group for raw string content (may be None/empty)
+        :param regular_capture: The capture group for regular string content (may be None/empty)
+        :returns: The actual string content
+        """
+        return raw_capture if raw_capture else regular_capture
+
     def find_strings(self) -> List[LocalizedString]:
         """Find all matching localized calls with a key specified in the buffer.
 
         :returns: The list of localized strings
         """
+        # The first thing to do is make sure there are no invalid calls to the function
+        self.confirm_string_args_only()
 
-        return self._detect_strings(
-            [
-                (SwiftDetector.LOCALIZED_PATTERN, 2),
-                (SwiftDetector.LOCALIZED_EXTENSION_PATTERN, 3),
-                (SwiftDetector.LOCALIZED_BUNDLE_PATTERN, 3),
-                (SwiftDetector.LOCALIZED_EXTENSION_BUNDLE_PATTERN, 4),
-            ],
-            SwiftDetector.LOCALIZED_BUNDLE_PATTERN,
-        )
+        # Find occurrences of `Localized` function calls
+        matches_in_buffer: List[Tuple] = []
+        patterns = [
+            (SwiftDetector.LOCALIZED_PATTERN, 4),  # 2 strings * 2 capture groups each
+            (SwiftDetector.LOCALIZED_EXTENSION_PATTERN, 6),  # 3 strings * 2 capture groups each
+            (SwiftDetector.LOCALIZED_BUNDLE_PATTERN, 6),  # 3 strings * 2 capture groups each
+            (SwiftDetector.LOCALIZED_EXTENSION_BUNDLE_PATTERN, 8),  # 4 strings * 2 capture groups each
+        ]
+
+        for pattern, count in patterns:
+            matches_in_buffer += self._get_matches(pattern, count)
+
+        results = []
+
+        for found_match in matches_in_buffer:
```

Comment:
Honestly, this loop is quite difficult to follow. Is there a way we can break this down into multiple PRs?